pkg/support/dump_test.go

```diff
@@ -38,3 +146,73 @@ func TestParseFileName(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, ts.String(), "2021-08-17 09:47:58 +0000 UTC")
 }
+
+func TestTimestampFilter(t *testing.T) {
+	ts := timestampFilter("2m")
+	currentTime := time.Now()
+	require.NotNil(t, ts)
+	expectedTime := currentTime.Add(-2 * time.Minute)
+	assert.WithinDuration(t, expectedTime, *ts, 10*time.Second)
+}
+
+func TestWriteFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "networkpolicies"
+	data := []byte("antct -oyaml get networkpolicies")
+	fs := afero.NewMemMapFs()
+	err := writeFile(fs, filepath.Join(basedir, resource), resource, data)
+	require.NoError(t, err)
+}
+
+func TestWriteYAMLFile(t *testing.T) {
+	basedir := "dir1/dir2"
+	resource := "agentinfo"
+	data := []byte("antct -oyaml get networkpolicies")
+	fs := afero.NewMemMapFs()
+	err := writeYAMLFile(fs, filepath.Join(basedir, resource), resource, data)
+	require.NoError(t, err)
+}
+
+func TestDumpAntctlGet(t *testing.T) {
+	basedir := "dir1/dir2"
+	name := "agentinfo"
+	exe := new(testExecuter)
+	fs := afero.NewMemMapFs()
+	err := dumpAntctlGet(fs, exe, name, basedir)
+	require.NoError(t, err)
```

Comment:
Going back to the comment I left for `func (te *testExecuter) Command` earlier.

In this test, I would expect you to check the contents of the `dir1/dir2/agentinfo` file in the `fs` file system. They should be equal to `antctl -oyaml get agentinfo`, unless I am missing something.

That would be the whole point of outputing the command name + args in `func (te *testExecuter) Command`: you want to be able to check that the correct command was invoked.

The same applies to all functions under test which use the `executor`.